Src/Newtonsoft.Json/JsonConvert.cs

```diff
@@ -962,9 +962,15 @@ public static void PopulateObject(string value, object target, JsonSerializerSet
             {
                 jsonSerializer.Populate(jsonReader, target);
 
-                if (jsonReader.Read() && jsonReader.TokenType != JsonToken.Comment)
+                if (settings != null && settings.CheckAdditionalContent)
```

Comment:
Adding this check because it seems that the current behaviour of throwing when there isn't a setting requesting it is, given that such a setting exists and is available to the user, a bug in itself. Maybe I'm misinterpreting the intent here.